sample-viewmodel/src/main/java/com/example/molecule/viewmodel/presentationLogic.kt

```diff
@@ -31,22 +31,40 @@ sealed interface Event {
 data class Model(
   val loading: Boolean,
   val breeds: List<String>,
-  val dropdownText: String,
+  val currentBreed: String?,
   val currentUrl: String?,
-)
-
-class PupperPicsViewModel : MoleculeViewModel<Event, Model>() {
-  @Composable
-  override fun models(events: Flow<Event>): Model {
-    return PupperPicsPresenter(events, PupperPicsService())
-  }
+) {
+  val dropdownText: String = currentBreed ?: "Select breed"
 }
 
+class PupperPicsViewModel(
+  // This service would typically be injected
+  service: PupperPicsService = PupperPicsService(),
+) : MoleculeViewModel<Event, Model>(
+  initialState = Model(
+    loading = false,
+    breeds = emptyList(),
+    currentBreed = null,
+    currentUrl = null,
+  ),
+  presenter = { seed, events ->
+    PupperPicsPresenter(
+      seed = seed,
+      events = events,
+      service = service,
+    )
+  },
+)
+
 @Composable
-fun PupperPicsPresenter(events: Flow<Event>, service: PupperPicsService): Model {
-  var breeds: List<String> by remember { mutableStateOf(emptyList()) }
-  var currentBreed: String? by remember { mutableStateOf(null) }
-  var currentUrl: String? by remember { mutableStateOf(null) }
+fun PupperPicsPresenter(
+  seed: Model,
```

Comment:
This `seed` thing sure is awkward. There "should" be no need for it — as a practical matter, `moleculeFlow` will emit immediately. But obviously that is not something a `Flow` does...

Reply:
The need for it mostly stems from restarting. Consider a timer that counts how much time is spent on a screen doing an activity, like the NYT crossword. It has the following presenter Composable:

```kotlin
@Composable
fun TimerPresenter(seed: Long): Long {
    var elapsed by remember { mutableStateOf(seed) }

    LaunchedEffect(Unit) {
        delay(1.seconds)
        ++elapsed
    }
}
```

Without the seed, there is no way to restore elapsed to its last seen state in between presenter stop and starts save for writing it to a data source. If the data source exposed a cold flow, the last seen value will immediately be overwritten:

```kotlin
@Composable
fun TimerPresenter(dataSource: Flow<Long>): Long {
    // There is no good default to use here upon resumption
    val savedElapsed by dataSource.collectAsState(0)
    var elapsed by remember { mutableStateOf(savedElapsed) }

    LaunchedEffect(Unit) {
        delay(1.seconds)
        ++elapsed
    }
}
```

In the above, the UI will briefly flash `0` when the screen is returned to after state production is stopped. The seed provides a way of "freezing" the last seen state for restoration when state production resumes as Compose Presenters unlike `Flow` require a synchronous (in as much as a Composable function can be deemed synchronous) initial return value.

Reply:
This isn't an issue with `Flow` operators like `combine` or `merge` because state won't be updated till all `Flows` emit in the case of the former, and till independent `Flows` emit in the latter. The immediate value requirement that Compose imposes seemingly mandates a seed value.

Reply:
I don't understand the `TimerPresenter` example. It seems to only emit one value.
```kotlin
LaunchedEffect(Unit) {
    delay(1000L)
    ++elapsed
}
```
Should it be inside a `while (true)` loop?

Reply:
Yes, it should. Thanks for pointing it out!

Reply:
If we do end up using a `while (true)` in the presenter and convert the molecule flow with `stateIn` and  `SharingStarted.WhileSubscribed(5.seconds)` I think the behavior would be wrong. In the sense that we would be tracking an additional 5 seconds after the last collector has been removed.
Would it be better to use molecule to create a StateFlow that is always active and using events to start and stop the timer be a better way to keep track of screen time?

Reply:
In the particular case of this example, a no argument `SharingStarted.WhileSubscribed()` would have been used. The example is a bit trivial however, as other flows may also have been used in the presenter.

The larger point in context being, it would be nice if molecule presenters could be used to create a `StateFlow` that offers behaviors that match that in all permutations of `flow.stateIn(...)`. Especially in the case where the `Flow` is stopped and restarted, without overwriting the last emitted value.